evm_rpc_types/src/response/mod.rs

```diff
@@ -126,3 +126,88 @@ pub struct TransactionReceipt {
     #[serde(rename = "type")]
     pub tx_type: HexByte,
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, CandidType)]
+pub struct Block {
+    /// Base fee per gas
+    /// Only included for blocks after the London Upgrade / EIP-1559.
+    #[serde(rename = "baseFeePerGas")]
+    pub base_fee_per_gas: Option<Nat256>,
+
+    /// Block number
+    pub number: Nat256,
+
+    /// Difficulty
+    pub difficulty: Option<Nat256>,
+
+    /// Extra data
+    #[serde(rename = "extraData")]
+    pub extra_data: Hex,
+
+    /// Maximum gas allowed in this block
+    #[serde(rename = "gasLimit")]
+    pub gas_limit: Nat256,
+
+    /// Gas used by all transactions in this block
+    #[serde(rename = "gasUsed")]
+    pub gas_used: Nat256,
+
+    /// Block hash
+    pub hash: Hex32,
+
+    /// Bloom filter for the logs.
+    #[serde(rename = "logsBloom")]
+    pub logs_bloom: Hex256,
+
+    /// Miner
+    pub miner: Hex20,
+
+    /// Mix hash
+    #[serde(rename = "mixHash")]
+    pub mix_hash: Hex32,
+
+    /// Nonce
+    pub nonce: Nat256,
```

Comment:
According to the [specification](https://github.com/ethereum/execution-apis/blob/1d4f70e84191bb574286fd7cea6c48795bf73e78/src/schemas/block.yaml#L70) this should be 8 bytes encoding as an hex string. I initially added the type `Hex8` (similar to the other `HexN` types); however this is not compatible with the current Candid API which declares `nonce` as a `nat` (whereas `HexN` is a subtype of `text`). We could introduce the type `Nat64` but this seems hardly worth it (this field was used for PoW blocks). WDYT?

Reply:
Using `Nat256` seems like a reasonable way to address this.